lighthouse-core/gather/gatherers/anchor-elements.js

```diff
@@ -87,8 +112,22 @@ class AnchorElements extends Gatherer {
       return (${collectAnchorElements})();
     })()`;
 
-    /** @type {Array<LH.Artifacts.AnchorElement>} */
-    return driver.evaluateAsync(expression, {useIsolation: true});
+    /** @type {LH.Artifacts['AnchorElements']} */
+    const anchors = await driver.evaluateAsync(expression, {useIsolation: true});
+    await driver.sendCommand('DOM.enable');
+
+    // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds.
```

Comment:
ah that sucks, `getDocument` is the hack to invoke `EnableAndReset` when `DOM.enable` was already called :/

https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/inspector/inspector_dom_agent.cc;l=462
```suggestion
    // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds if the `DOM` domain was enabled before this gatherer, invoke it to be safe.
```

Reply:
Ah yeah I saw it elsewhere and copied it along with that comment. Without it, it seemed I couldn't resolve nodes, they weren't being found. Anyway good comment suggestion, added. And really interesting, finding that in the chromium source, kinda cool to see